contracts/carbon/Strategies.sol

```diff
@@ -972,6 +977,13 @@ abstract contract Strategies is Initializable {
         }
     }
 
+    function tryAdd(uint256 x, uint256 y) private pure returns (bool safe, uint256 sum) {
```

Comment:
Since you aren't using the `sum` in case `safe` is `false`, you can use OZ's `tryAdd` explicitly: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e86bb45477d335294086c333047626b7fd317e5a/contracts/utils/math/Math.sol#L25

Reply:
Yeah, I initially did, but for some reason, it makes the size of the test (helper) contract exceed the maximum, and I wanted to avoid the compilation warning.
The second reason is that we actually have a few other things in there, which we could technically replace with OZ code, so it didn't seem important to follow a line which we aren't enforcing anyway.
P.S.: if OZ had given it just a little bit more thought, then they would have probably ended up doing it the exact same way we did it here.

Reply:
The size of test contracts isn't really an issue. They will eventually become too large, even without the fix. DIY or code duplicate is always undesired. If there are other places like this, IMHO they should be refactored just as well. Less DIY/duplication == easier to review and reason about the code.

@yudilevi your call

Reply:
We also don't use OZ's `mulDiv` (with `Rounding.Up` and `Rounding.Down`), which they've apparently added at some point, and can replace our `mulDivF` and `mulDivC` (all across the code of course, otherwise, we get two different versions of each one of these functions).

Reply:
I’d probably go with OZ’s code for any new code (no muldiv replacement yet)

Reply:
OK, please note that OZ's implementation is slightly more expensive than ours:
- OZ implementation: 131 gas in case of an overflow and 117 gas in case of no overflow
- Our implementation: 38 gas regardless of whether an overflow occurs or doesn't occur

This means:
- 79 gas **per order** per trade in most cases (those in which an overflow doesn't occur)
- 93 gas **per order** per trade in rare cases (those in which an overflow does occur)

Not significant but still something to consider...

Reply:
IMHO, insignificant and can be ignored.

Reply:
Done